scrapy_poet/downloadermiddlewares.py

```diff
@@ -146,11 +146,14 @@ def process_response(
             request,
             response,
         )
-        # Fill the callback arguments with the created instances
+        self._merge_dependencies(final_kwargs, request)
+        return response
+
+    @staticmethod
+    def _merge_dependencies(final_kwargs: Dict, request: Request) -> None:
```

Comment:
I think we can remove this method, as it doesn't really merge the user-provided dependencies anymore